crates/journal/src/journal.rs

```diff
@@ -137,11 +148,10 @@ pub fn new_journal_entry(app_state: Arc<AppState>, cx: &mut WindowContext) {
 }
 
 fn journal_dir(path: &str) -> Option<PathBuf> {
-    let expanded_journal_dir = shellexpand::full(path) //TODO handle this better
+    let expanded_path = shellexpand::full(path)
         .ok()
-        .map(|dir| Path::new(&dir.to_string()).to_path_buf().join("journal"));
```

Comment:
Previously with no journal key under settings (the default) spawning the `new journal entry` action would create the `~/journal/%Y/%M/` directory and open a new workspace at `~/journal`.  For me, this branch spawned a workspace at `~` which immediately triggered a bunch of MacOS folder security prompts as Zed tried to recursively traverse my home dir.

I think perhaps the `journal.path` default should be updated to be `~/journal`:
https://github.com/zed-industries/zed/blob/e64def052b166bb6f314f4d55516d0446ab865c2/assets/settings/default.json#L554-L557

Can you ensure the behavior remains the same with no settings?